files/bootstrap.sh

```diff
@@ -455,11 +455,23 @@ if [[ "$USE_MAX_PODS" = "true" ]]; then
   echo "$(jq ".maxPods=$MAX_PODS" $KUBELET_CONFIG)" > $KUBELET_CONFIG
 fi
 
+KUBELET_ARGS="--node-ip=$INTERNAL_IP --pod-infra-container-image=$PAUSE_CONTAINER --v=2"
+
+if vercmp "$KUBELET_VERSION" lt "1.26.0"; then
+  # TODO: remove this when 1.25 is EOL
+  KUBELET_CLOUD_PROVIDER="aws"
+else
+  KUBELET_CLOUD_PROVIDER="external"
+  echo "$(jq ".providerID=\"$(provider-id)\"" $KUBELET_CONFIG)" > $KUBELET_CONFIG
```

Comment:
How does node object look like when provider id isn't passed here?

Reply:
The `Node`'s `spec.providerID` would be missing. The cloud provider works around this by resolving the node name to an instance ID, i.e.:

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller.go#LL219C26-L219C26

calls `instances.InstanceID`, which calls:

https://github.com/kubernetes/cloud-provider-aws/blob/master/pkg/providers/v1/aws.go#L5125

If kubelet doesn't specify the provider ID, it's set using this approach before the node's uninitialized taint is removed: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go#L527-L528C4